packages/compiler/src/render3/partial/api.ts

```diff
@@ -371,6 +371,48 @@ export enum R3FactoryTarget {
   NgModule = 4,
 }
 
+/**
+ * Describes the shape of the object that the `ɵɵngDeclareInjectable()` function accepts.
+ *
+ * This interface serves primarily as documentation, as conformance to this interface is not
+ * enforced during linking.
+ */
+export interface R3DeclareInjectableMetadata extends R3PartialDeclaration {
+  /**
+   * If provided, specifies that the declared injecteble belongs to a particular injector:
+   * - `InjectorType` such as `NgModule`,
+   * - `'root'` the root injector
+   * - `'any'` all injectors.
+   * - `null`, does not belong to any injector. Must be explicitly listed in the injector
+   * `providers`.
+   */
+  providedIn?: o.Expression;
+  /**
+   * If provided, an expression that evaluates to a class to use when creating an instance of this
+   * injectable.
+   */
+  useClass?: o.Expression;
+  /**
+   * If provided, an expression that evaluates to a function to use when creating an instance of
+   * this injectable.
+   */
+  useFactory?: o.Expression;
+  /**
+   * If provided, an expression that evaluates to a token of another injectable that this injectable
+   * aliases.
+   */
+  useExisting?: o.Expression;
+  /**
+   * If provided, an expression that evaluates to the value of the instance of this injectable.
+   */
+  useValue?: o.Expression;
+  /**
+   * An array of dependencies to support instantiating this injecttable via `useClass` or
```

Comment:
```suggestion
   * An array of dependencies to support instantiating this injectable via `useClass` or
```

Looks like "injectable" is a hard word :-)

Reply:
Oh Man! It is nearly as hard as ~~unkown~~ ~~uknown~~ unknown.